ct_only.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tessera
+
+import (
+	"context"
+
+	"github.com/transparency-dev/trillian-tessera/ctonly"
+)
+
+// Storage described the expected functions from Tessera storage implementations.
+type Storage interface {
+	// Add should duably assign an index to the provided Entry, and return it.
+	//
+	// Implementations MUST call MarshalBundleData method on the entry before marshaling and persisting it.
```

Comment:
This now reads a bit funny because we must call the marshaling function before marshaling it? Maybe just remove the second marshaling?

Reply:
d'oh, fixed - ta.